jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -813,9 +814,20 @@ private static void ensureProperties(boolean isInitialization) {
 private static void initializeSystemProperties(Map<String, String> mapProperties) {
 	systemProperties = new Properties();
 	for (Map.Entry<String, String> property : mapProperties.entrySet()) {
-		systemProperties.put(property.getKey(), property.getValue());
+		String key = property.getKey();
+		/* Remove OpenJDK private properties that should not be System properties. */
+		switch (key) {
+			case "sun.nio.MaxDirectMemorySize": //$NON-NLS-1$
+			case "sun.nio.PageAlignDirectMemory": //$NON-NLS-1$
+			case "java.lang.Integer.IntegerCache.high": //$NON-NLS-1$
+			case "sun.java.launcher.diag": //$NON-NLS-1$
+			case "jdk.boot.class.path.append": //$NON-NLS-1$
```

Comment:
Please list the cases alphabetically and add an explicit `default` case.

Reply:
Also, by convention, each `case` should be indented the same as `switch`.

Reply:
Addressed all the review comments.